### PR TITLE
Add visually-hidden text for rejection reasons fields

### PR DESCRIPTION
**Pull request by frankieroberto**
This adds visually-hiddden text to the otherwise-identical labels and legends within the rejections reasons interface, so that screen reader users can differentiate them when they are announced out of context.

### Non-unique legends

On the reasons for rejection page, there is a list of checkboxes, most of which when clicked reveal a nested set of checkboxes with the heading "Reasons" (implemented as a legend):

Screenshot 2023-05-11 at 16 16 39

Because these legends all have the same text, this may be confusing or difficult for users of screen-reading software, which can announce these legends out of the context of the previous checkbox label.

To resolve this, visually-hidden text can be added to the legends, so that they contain "Reasons [about why their qualifications do not meet course requirements]".

### Non-unique textarea labels

When the nested checkboxes are checked, most of them reveal a textarea labelled "Details":

Screenshot 2023-05-11 at 16 25 35

Because these legends all have the same text, this may be confusing or difficult for users of screen-reading software or dictation software to enter text.

To resolve this, visually-hidden text can be added to the labels, so that they contain "Details [about why you could not verify qualifications]".

🗂️ [Trello card](https://trello.com/c/sFGCwJeI/1438-sr4r-add-visually-hidden-text-to-form-elements)

### CODE REVIEW

**Comment by frankieroberto** [comment]
[SR4R - Add visually hidden text to form elements](https://trello.com/c/sFGCwJeI/1438-sr4r-add-visually-hidden-text-to-form-elements)

File: config/rejection_reasons.yml [hunk 1]
```diff
@@ -16,85 +17,103 @@
...
 - :id: personal_statement
   :label: Personal statement
+  :reasons_visually_hidden: for rejecting due to personal statement
   :reasons:
   - :id: quality_of_writing
     :label: Quality of writing
     :details:
       :id: quality_of_writing_details
       :label: Details
+      :visually_hidden: about their quality of writing
```
Comment (MaeveRoseveare) [inline config/rejection_reasons.yml:42]:
```suggestion
      :visually_hidden: about their quality of writing in their personal statement
```

File: config/rejection_reasons.yml [hunk 2]
```diff
@@ -16,85 +17,103 @@
...
 - :id: teaching_knowledge
   :label: Teaching knowledge, ability and interview performance
+  :reasons_visually_hidden: for rejecting due to teaching knowledge or interview performance
   :reasons:
   - :id: subject_knowledge
     :label: Subject knowledge
     :details:
       :id: subject_knowledge_details
       :label: Details
+      :visually_hidden: about their subject knowledge
```
Comment (MaeveRoseveare) [inline config/rejection_reasons.yml:58]:
```suggestion
      :visually_hidden: about issues with their subject knowledge
```

File: config/rejection_reasons.yml [hunk 3]
```diff
@@ -16,85 +17,103 @@
...
   - :id: safeguarding_knowledge
     :label: Safeguarding knowledge
     :details:
       :id: safeguarding_knowledge_details
       :label: Details
+      :visually_hidden: about their safeguarding knowledge
```
Comment (MaeveRoseveare) [inline config/rejection_reasons.yml:64]:
```suggestion
      :visually_hidden: about issues with their safeguarding knowledge
```

File: config/rejection_reasons.yml [hunk 4]
```diff
@@ -16,85 +17,103 @@
...
   - :id: teaching_method_knowledge
     :label: Teaching method knowledge
     :details:
       :id: teaching_method_knowledge_details
       :label: Details
+      :visually_hidden: about their teaching method knowledge
```
Comment (MaeveRoseveare) [inline config/rejection_reasons.yml:70]:
```suggestion
      :visually_hidden: about issues with their teaching method knowledge
```

File: config/rejection_reasons.yml [hunk 5]
```diff
@@ -16,85 +17,103 @@
...
   - :id: teaching_role_knowledge
     :label: Teaching role knowledge
     :details:
       :id: teaching_role_knowledge_details
       :label: Details
+      :visually_hidden: about their teaching role knowledge
```
Comment (MaeveRoseveare) [inline config/rejection_reasons.yml:76]:
```suggestion
      :visually_hidden: about issues with their teaching role knowledge
```

File: config/rejection_reasons.yml [hunk 6]
```diff
@@ -16,85 +17,103 @@
...
   - :id: teaching_demonstration
     :label: Teaching demonstration
     :details:
       :id: teaching_demonstration_details
       :label: Details
+      :visually_hidden: about their teaching demonstration
```
Comment (MaeveRoseveare) [inline config/rejection_reasons.yml:82]:
```suggestion
      :visually_hidden: about issues with their teaching demonstration
```

File: config/rejection_reasons.yml [hunk 7]
```diff
@@ -16,85 +17,103 @@
...
     :details:
       :id: did_not_reply_details
       :label: Details
+      :visually_hidden: of how they did not reply to messages
   - :id: did_not_attend_interview
     :label: Did not attend interview
     :details:
       :id: did_not_attend_interview_details
       :label: Details
+      :visually_hidden: of how they did not attend interview
```
Comment (MaeveRoseveare) [inline config/rejection_reasons.yml:104]:
```suggestion
      :visually_hidden: of how they did not attend interviews
```

File: config/rejection_reasons.yml [hunk 8]
```diff
@@ -106,19 +125,23 @@
...
     :id: course_full_details
     :label: Details (optional)
     :optional: true
+    :visually_hidden: about the course being full
 - :id: other
   :label: Other
   :details:
     :id: other_details
     :label: Details
+    :visually_hidden: of other reasons
```
Comment (MaeveRoseveare) [inline config/rejection_reasons.yml:147]:
```suggestion
    :visually_hidden: of other reasons for rejection
```

**Review by MaeveRoseveare — APPROVED** [review]
Looks, good made a few tweaks!

**Review by stevehook — APPROVED** [review]
:+1: Makes sense and works for me locally

**Pull request merged**